app/stac_api/views.py

```diff
@@ -54,3 +60,30 @@ def checker(request):
     data = {"success": True, "message": "OK"}
 
     return JsonResponse(data)
+
+
+class CollectionList(generics.ListAPIView):
+    '''
+    collections endpoint
+    '''
+    # TODO: restrict the visible fields in ListView if possible # pylint: disable=fixme
+    # seems to be tricky. Would need dynamic serializers or different serializers
+    # one for ListView and one for the DetailView..
+    # working on it..
+    serializer_class = CollectionSerializer
+    queryset = Collection.objects.all()
+
+
+class CollectionDetail(generics.RetrieveAPIView):
+    '''
+    Returns a detail view of the collection instance with all its properties
+    '''
+    serializer_class = CollectionSerializer
+    lookup_url_kwarg = "collection_name"
+    queryset = Collection.objects.all()
+
+    def get_object(self):
+        collection_name = self.kwargs.get(self.lookup_url_kwarg)
+        queryset = self.get_queryset().filter(collection_name=collection_name)
```

Comment:
Just a small coding style comment here. I would put these class into a collection.py module and put two functions in views.py as follow:

```python
def collection_list():
    return CollectionList.as_view()
```

This would keep views.py concise as we add endpoints, otherwise this file will become too big. Thg views.py import gets also simplified as the `from . import views` is sufficient.

Reply:
Good idea, did that